src/bundle/Core/Command/VirtualFieldDuplicateFixCommand.php

```diff
@@ -0,0 +1,272 @@
+            'batch-size',
+            'b',
+            InputOption::VALUE_OPTIONAL,
+            'Number of attributes affected per iteration',
+            self::DEFAULT_BATCH_SIZE
+        );
+
+        $this->addOption(
+            'max-iterations',
+            'i',
+            InputOption::VALUE_OPTIONAL,
+            'Max iterations count (default or -1: unlimited)',
+            self::MAX_ITERATIONS_UNLIMITED
+        );
+
+        $this->addOption(
+            'sleep',
+            's',
+            InputOption::VALUE_OPTIONAL,
+            'Wait between iterations, in milliseconds',
+            self::DEFAULT_SLEEP
+        );
+
+        $this->addOption(
+            'force',
+            'f',
+            InputOption::VALUE_NONE,
+            'Force operation (implies non-interactive mode)',
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $style = new SymfonyStyle($input, $output);
+        $stopwatch = new Stopwatch(true);
+        $stopwatch->start('total', 'command');
+
+        $force = $input->getOption('force');
+        if ($force) {
+            $input->setInteractive(false);
+        }
+
+        $batchSize = (int)$input->getOption('batch-size');
+        if ($batchSize === 0) {
+            $style->warning('Batch size is set to 0. Nothing to do.');
+
+            return Command::INVALID;
+        }
+
+        $maxIterations = (int)$input->getOption('max-iterations');
+        if ($maxIterations === 0) {
+            $style->warning('Max iterations is set to 0. Nothing to do.');
+
+            return Command::INVALID;
+        }
+
+        $sleep = (int)$input->getOption('sleep');
+
+        try {
+            $totalCount = $this->getDuplicatedAttributeTotalCount($style, $stopwatch);
+
+            if ($totalCount > 0) {
+                $confirmation = $this->askForConfirmation($style);
+                if (!$confirmation && !$force) {
+                    $style->info('Confirmation rejected. Terminating.');
+
+                    return Command::FAILURE;
+                }
+            } else {
+                $style->success('Database is clean of attribute duplicates. Nothing to do.');
+
+                return Command::SUCCESS;
+            }
+
+            $iteration = 1;
+            $totalDeleted = 0;
+            do {
+                $deleted = 0;
+                $stopwatch->start('iteration', 'sql');
+
+                $attributes = $this->getDuplicatedAttributesBatch($batchSize);
+                foreach ($attributes as $attribute) {
+                    $attributeIds = $this->getDuplicatedAttributeIds($attribute);
+
+                    $deleted += $this->deleteAttributes($attributeIds);
+                    $totalDeleted += $deleted;
+                }
+
+                $style->info(
+                    sprintf(
+                        'Iteration %d: Removed %d duplicates (total removed this execution: %d). [Debug %s]',
```

Comment:
This message could be clearer. 🙂 At this moment it's unclear which is duplicated lines and which attributes.

Reply:
Message refers to removed database rows (there was issue in counting elsewhere and made it look like duplicates and total duplicates refers to different things).